File: app/src/main/java/org/wikipedia/page/PageBackStackItem.java
package org.wikipedia.page;

import androidx.annotation.NonNull;

import org.wikipedia.history.HistoryEntry;

public class PageBackStackItem {
    @NonNull private PageTitle title;
    @NonNull private HistoryEntry historyEntry;

    private int scrollY;

    public PageBackStackItem(@NonNull PageTitle title, @NonNull HistoryEntry historyEntry) {

        // TODO: remove this crash probe upon fixing
        if (title == null || historyEntry == null) {
            throw new IllegalArgumentException("Nonnull parameter is in fact null.");
        }

        this.title = title;
        this.historyEntry = historyEntry;
    }

    @NonNull
    public PageTitle getTitle() {
        return title;
    }

    public void setTitle(@NonNull PageTitle title) {
        this.title = title;
    }

    @NonNull
    public HistoryEntry getHistoryEntry() {
        return historyEntry;
    }

    public int getScrollY() {
        return scrollY;
    }

    public void setScrollY(int scrollY) {
        this.scrollY = scrollY;
    }
}

File: app/src/main/java/org/wikipedia/page/PageBackStackItem.kt
package org.wikipedia.page

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable
import org.wikipedia.history.HistoryEntry
import java.util.Date

@Serializable
@Entity
class PageBackStackItem(
    @PrimaryKey(autoGenerate = true) var id: Long = 0,
    val apiTitle: String,
    val displayTitle: String,
    val langCode: String,
    val namespace: String,
    val timestamp: Long = Date().time,
    var scrollY: Int = 0
) {
    @Ignore
    var title: PageTitle? = null

    @Ignore
    var historyEntry: HistoryEntry? = null
}